### PR TITLE
#1667 refactor config includes 

### LINKED ISSUE #1667: Explore refactoring includes in config to minimize and simplfiy inclusions
**What Needs to be Done?**

Motivated by this PR: https://github.com/DARMA-tasking/vt/pull/1658

### PR DESCRIPTION
**Pull request by jstrzebonski**
Fixes #1667

### CODE REVIEW

File: src/vt/configs/error/pretty_print_stack.h [hunk 1]
```diff
@@ -45,8 +45,7 @@
 #define INCLUDED_VT_CONFIGS_ERROR_PRETTY_PRINT_STACK_H
 
 #include "vt/config.h"
-#include "vt/configs/error/stack_out.h"
-#include "vt/configs/debug/debug_colorize.h"
+#include "vt/context/context.h"
 
 #include <string>
```
Comment (PhilMiller) [inline src/vt/configs/error/pretty_print_stack.h:51]: Is there any reason this needs to be `inline` and defined in a header? It's purely debug code. We should move the implementation to a cc file, and then can move the include of `context.h` there.

File: tests/perf/common/test_harness.cc [hunk 2]
```diff
@@ -52,7 +52,13 @@
 #include <numeric>
 #include <fstream>
 
-namespace vt { namespace tests { namespace perf { namespace common {
+namespace vt {
+
+namespace debug {
+arguments::AppConfig *preConfigRef();
+} // namespace debug
```
Comment (PhilMiller) [inline tests/perf/common/test_harness.cc:59]: This repeated forward declaration in several .cc files is a little smelly

Comment (PhilMiller) [reply tests/perf/common/test_harness.cc:59]: For the few places this occurs, I think it would be reason to include a common header

Comment (jstrzebonski) [reply tests/perf/common/test_harness.cc:59]: Maybe I over-refactored this part of the code. Let me put it back to where it was.

File: src/vt/configs/error/pretty_print_stack.h (hunk 1 above)
Comment (jstrzebonski) [reply src/vt/configs/error/pretty_print_stack.h:50]: Done.

File: tests/perf/common/test_harness.cc (hunk 2 above)
Comment (jstrzebonski) [reply tests/perf/common/test_harness.cc]: Done.

File: src/vt/configs/error/pretty_print_stack.h [hunk 3]
```diff
@@ -44,55 +44,13 @@
...
-    );
-    out += ret_str;
-    i++;
-  }
-
-  //out += seperator + seperator + seperator;
-
-  return out;
-}
+std::string prettyPrintStack(StackVectorType const& stack);
```
Comment (PhilMiller) [inline src/vt/configs/error/pretty_print_stack.h:53]: Could this just live in `stack_out.h`, and eliminate this header entirely?

File: src/vt/configs/error/pretty_print_stack.cc [hunk 4]
```diff
@@ -41,11 +41,54 @@
...
-#endif /*INCLUDED_VT_CONFIGS_FEATURES_FEATURES_ENABLEIF_H*/
+std::string prettyPrintStack(StackVectorType const& stack) {
+  auto green      = ::vt::debug::green();
+  auto red        = ::vt::debug::red();
+  auto bred       = ::vt::debug::bred();
+  auto reset      = ::vt::debug::reset();
+  auto bd_green   = ::vt::debug::bd_green();
+  auto magenta    = ::vt::debug::magenta();
+  auto blue       = ::vt::debug::blue();
+  auto yellow     = ::vt::debug::yellow();
```
[the remark this answers is not in the document]
Comment (jstrzebonski) [reply src/vt/configs/error/pretty_print_stack.cc]: Fixed.

File: src/vt/configs/error/pretty_print_stack.h (hunk 3 above)
Comment (jstrzebonski) [reply src/vt/configs/error/pretty_print_stack.h]: Done.

**Review by PhilMiller — APPROVED** [review]

**Pull request merged**